### PR TITLE
Reduce logging level for transient job processing issues.

### PR DESCRIPTION
**Pull request by isoos**
https://github.com/dart-lang/pub-dartlang-dart/issues/1604

### CODE REVIEW

File: app/lib/job/job.dart [hunk 1]
```diff
@@ -55,8 +55,12 @@ abstract class JobProcessor {
             _logger.info('$_serviceAsString job completed: $jobDescription');
           } catch (e, st) {
             status = JobStatus.failed;
-            _logger.severe(
-                '$_serviceAsString job error $jobDescription', e, st);
+
+            final str = '$e';
```
Comment (sortie) [inline app/lib/job/job.dart]: e.toString() might be faster / more direct but this is fine.

File: app/lib/job/job.dart [hunk 2]
```diff
@@ -55,8 +55,12 @@ abstract class JobProcessor {
...
+            final isTransient = str.contains('The transaction was aborted.');
```
Comment (sortie) [inline app/lib/job/job.dart]: There isn't a way to catch this exception a particular class?

Comment (isoos) [reply app/lib/job/job.dart]: > There isn't a way to catch this exception a particular class?

Actually, we can :) PTAL.

File: app/lib/job/job.dart [hunk 3]
```diff
@@ -48,13 +50,16 @@ abstract class JobProcessor {
...
+          JobStatus status = JobStatus.failed;
           try {
             status = await process(job);
-            status ??= JobStatus.success;
+            status = JobStatus.success;
             _logger.info('$_serviceAsString job completed: $jobDescription');
+          } on TransactionAbortedError catch (e, st) {
+            _logger.info('$_serviceAsString job error $jobDescription', e, st);
+          } on TimeoutError catch (e, st) {
+            _logger.info('$_serviceAsString job error $jobDescription', e, st);
```
Comment (sortie) [inline app/lib/job/job.dart]: Maybe add a "(retrying later)" to the message if it is retried, so it's obvious it's harmless?

Comment (isoos) [reply app/lib/job/job.dart]: I think it will be redundant, because it will be retried in all of the cases, regardless of the error type.

**Pull request merged**